src/test/java/com/mysql/cj/jdbc/NonRegisteringDriverTest.java

```diff
@@ -37,16 +37,36 @@
 import org.mockito.MockedStatic;
 import org.mockito.Mockito;
 
+import java.lang.reflect.Method;
 import java.sql.Connection;
 import java.sql.SQLException;
 import java.util.Properties;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
 import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.*;
 
 @SuppressWarnings("unchecked")
 public class NonRegisteringDriverTest {
+
+    @Test
+    public void testSetAwsProtocolOnlySwitch() throws Exception {
```

Comment:
I think this test needs a few more cases to cover the different combinations of driver-level and connection-level settings. I'll follow up directly on slack with you about it